starter-core/src/main/java/io/micronaut/starter/feature/buildless/Buildless.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2017-2021 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.starter.feature.buildless;
+
+import io.micronaut.starter.application.ApplicationType;
+import io.micronaut.starter.application.generator.GeneratorContext;
+import io.micronaut.starter.build.gradle.GradleFile;
+import io.micronaut.starter.build.gradle.GradlePlugin;
+import io.micronaut.starter.feature.Category;
+import io.micronaut.starter.feature.Feature;
+import io.micronaut.starter.feature.build.BuildCacheConfiguration;
+import io.micronaut.starter.feature.build.gradle.templates.buildlessGradlePlugin;
+import io.micronaut.starter.feature.build.gradle.templates.gradleBuildCache;
+import io.micronaut.starter.template.RockerWritable;
+import jakarta.inject.Singleton;
+
+@Singleton
+public class Buildless implements Feature, BuildCacheConfiguration {
+    public static final boolean BUILDLESS_ENABLE_PLUGIN = true;
+    public static final String FEATURE_NAME_BUILDLESS = "buildless";
+    public static final String BUILDLESS_PLUGIN_ID = "build.less";
+    public static final String BUILDLESS_PLUGIN_VERSION = "1.0.0-beta1";
```

Comment:
this should be defined in [tool file](https://github.com/micronaut-projects/micronaut-starter/blob/4.1.x/gradle/templates.versions.toml)